### ISSUE TITLE
Repackage fails to keep the timestamps of jar entries (JarWriter)

### ISSUE THREAD

**Issue by jccarrillo** [spring-projects/spring-boot#4767]
When running repackage, JarWriter fails to keep the original timestamp entries of libraries.
For instance, see chunk below:
```
  2245 Fri Dec 11 20:37:34 EST 2015 lib/spring-boot-starter-actuator-1.2.5.RELEASE.jar
  2215 Fri Dec 11 20:37:34 EST 2015 lib/spring-boot-starter-1.2.5.RELEASE.jar
  2247 Fri Dec 11 20:37:34 EST 2015 lib/spring-boot-starter-logging-1.2.5.RELEASE.jar
273594 Fri Dec 11 20:37:34 EST 2015 lib/spring-boot-actuator-1.2.5.RELEASE.jar
1052383 Fri Dec 11 20:37:34 EST 2015 lib/spring-core-4.2.0.RELEASE.jar
 18709 Wed Jul 01 22:20:40 EDT 2015 org/springframework/boot/loader/PropertiesLauncher.class
     0 Fri Dec 11 20:37:34 EST 2015 org/springframework/boot/loader/data/
  1531 Wed Jul 01 22:20:40 EDT 2015 org/springframework/boot/loader/data/ByteArrayRandomAccessData.class
  3526 Wed Jul 01 22:20:40 EDT 2015 org/springframework/boot/loader/data/RandomAccessDataFile$DataInputStream.class
   551 Wed Jul 01 22:20:40 EDT 2015 org/springframework/boot/loader/data/RandomAccessData.class
  2235 Wed Jul 01 22:20:40 EDT 2015 org/springframework/boot/loader/data/RandomAccessDataFile$FilePool.class
  3390 Wed Jul 01 22:20:40 EDT 2015 org/springframework/boot/loader/data/RandomAccessDataFile.class
...
```

It would be beneficial to keep timestamps.

I believe the fix would be in:
https://github.com/spring-projects/spring-boot/blob/master/spring-boot-tools/spring-boot-loader-tools/src/main/java/org/springframework/boot/loader/tools/JarWriter.java

**Comment by philwebb** [comment]
See #2807

**Comment by philwebb** [comment]
The timestamp should be set to the value of the first entry within the nested jar. The reason for this is that build tools seem to set the timestamp of jars as they are downloaded, so they aren't consistent across different machines. The ultimate goal of #2807 was to allow efficient use of docker filesystem layers.

Are you finding a specific problem with the current strategy? Perhaps we can add a flag if there is a compelling reason to keep the timestamp from the source jar file. What build tool are you using?

**Comment by philwebb** [comment]
Oh wait, looks like you're using v1.2, the #2807 fix was for v1.3+. Have you tried the latest release?

**Comment by jccarrillo** [comment]
I just tried it and it's been fixed in 1.3.0.RELEASE. I was using 1.2.5.RELEASE. Thanks!

```
  2245 Wed Jul 01 22:28:22 EDT 2015 lib/spring-boot-starter-actuator-1.2.5.RELEASE.jar
  2215 Wed Jul 01 22:27:50 EDT 2015 lib/spring-boot-starter-1.2.5.RELEASE.jar
  2247 Wed Jul 01 22:27:50 EDT 2015 lib/spring-boot-starter-logging-1.2.5.RELEASE.jar
 16618 Thu Mar 26 21:57:08 EDT 2015 lib/jcl-over-slf4j-1.7.12.jar
  4736 Thu Mar 26 21:57:32 EDT 2015 lib/jul-to-slf4j-1.7.12.jar
 24567 Thu Mar 26 21:57:18 EDT 2015 lib/log4j-over-slf4j-1.7.12.jar
273594 Wed Jul 01 22:27:44 EDT 2015 lib/spring-boot-actuator-1.2.5.RELEASE.jar
1052383 Fri Jul 31 01:33:06 EDT 2015 lib/spring-core-4.2.0.RELEASE.jar
  5503 Wed Jul 01 22:28:30 EDT 2015 lib/spring-boot-starter-remote-shell-1.2.5.RELEASE.jar
1026661 Tue Jun 30 09:45:02 EDT 2015 lib/spring-context-4.1.7.RELEASE.jar
155497 Fri Feb 06 10:47:12 EST 2015 lib/crash.cli-1.3.1.jar
 38114 Fri Feb 06 10:54:18 EST 2015 lib/crash.connectors.ssh-1.3.1.jar
```

**Comment by philwebb** [comment]
Thanks. I'm afraid it's a little bit too invasive to backport the fix, so it'll need to remain on 1.3+ only. Hopefully you can upgrade without too much issue.
